src/utils/positionUtils.ts

```diff
@@ -254,30 +254,38 @@ export const calcTradeFee = ({
   isMaker: boolean
   direction: PositionSide
 }) => {
-  const noValue = { total: 0n, impactFee: 0n, skewFee: 0n, feeBasisPoints: 0n }
+  const noValue = { total: 0n, feeBasisPoints: 0n, proportionalFee: 0n, linearFee: 0n, adiabaticFee: 0n }
   if (!marketSnapshot || !positionDelta) return noValue
 
-  // TODO (v2.2): Fix fee calcs
   const {
     riskParameter: { takerFee, makerFee },
-    nextPosition: { long, short, maker },
+    nextPosition: { long, short },
     global: { latestPrice },
+    makerTotal,
+    takerTotal,
   } = marketSnapshot
 
   const notional = calcNotional(positionDelta, latestPrice)
 
   if (isMaker) {
-    const major = Big6Math.max(long, short)
-    const minor = Big6Math.min(long, short)
-    const currentUtilization = maker + minor !== 0n ? Big6Math.div(major, maker + minor) : 0n
-    const newUtilization =
-      maker + minor + positionDelta !== 0n ? Big6Math.div(major, maker + minor + positionDelta) : 0n
-    const utilizationDelta = newUtilization - currentUtilization
-    const impactFee = Big6Math.mul(makerFee.proportionalFee, utilizationDelta)
-    const total = Big6Math.max(Big6Math.mul(notional, impactFee + makerFee.linearFee), 0n)
-    const feeBasisPoints = !Big6Math.isZero(total) ? Big6Math.div(total, notional) : makerFee.linearFee
-
-    return { impactFee, total, skewFee: undefined, feeBasisPoints }
+    const adjustedMakerTotal = makerTotal + Big6Math.abs(positionDelta)
+    const makerProportionalFeeRate = Big6Math.mul(
+      makerFee.proportionalFee,
+      Big6Math.div(adjustedMakerTotal, makerFee.scale),
+    )
+    const makerProportionalFee = Big6Math.mul(notional, makerProportionalFeeRate)
+    const makerLinearFee = Big6Math.mul(notional, makerFee.linearFee)
+    const total = makerLinearFee + makerProportionalFee
+    const feeBasisPoints = !Big6Math.isZero(total) ? Big6Math.div(total, notional) : 0n // Default to 0bps if fee total is 0?
+
+    return {
+      total,
+      feeBasisPoints,
+      proportionalFee: makerProportionalFee,
+      linearFee: makerLinearFee,
+      adiabaticFee: 0n,
+      impactFee: 0n,
```

Comment:
We can remove `impactFee` from this type, we shouldn't have that type of fee anymore